src/components/PopupWithForm.js

```diff
@@ -19,8 +20,11 @@ export class PopupWithForm extends Popup {
   setEventListeners() {
     this._popupForm.addEventListener("submit", (event) => {
       event.preventDefault();
+      const submitText = this._submitButton.textContent;
+      this._submitButton.textContent = "Saving...";
       this._data = this._getInputValues();
       this._handleSubmit(event, this._data);
+      this._submitButton.textContent = submitText;
     });
     super.setEventListeners();
   }
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // Here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`